app/src/main/java/com/gkp/tasky/MainActivity.kt

```diff
@@ -4,19 +4,36 @@ import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.enableEdgeToEdge
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
 import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
 import com.gkp.core.designsystem.theme.TaskyTheme
 import com.gkp.tasky.nagivation.TaskyNavHost
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class MainActivity : ComponentActivity() {
+
+    private val viewModel: MainViewModel by viewModel()
+    private var navHostStartDestinationSet by mutableStateOf(false)
```

Comment:
This is going to be reset after every config change. Is that what you want?

Reply:
No. This means I have to move it  to the ViewModel right?